File: Eulerpl/Euler2.pl
#!/usr/bin/perl

$previous = 0;
$current = 1;
$next = 1;
$sum = 0;
while($current < 4000000){
	$next = $previous + $current;
	$previous = $current;
	$current = $next;
	if($current % 2 == 0){
	$sum = $sum + $current;
} 
}	
print "$sum\n";

File: Eulercpp/Euler2.cpp
#include<iostream>
using namespace std;

int main(void) {
	int sum = 0;
	int current = 1;
	int next = 1;
	int temp = 1;
	while(current < 4000000) {
		if(current % 2 == 0){
			sum += current;
		}
		temp = next;
		next = next + current;
		current = temp;
	}
	cout << sum << endl;
}
